core/src/main/javascript/crypto/EccCryptoContext.js

```diff
@@ -33,12 +45,13 @@ var EccCryptoContext;
          * @param {PrivateKey} privateKey the private key.
          * @constructor
          */
-        init: function init(ctx, privateKey) {
+        init: function init(ctx, privateKey, publicKey, mode) {
```

Comment:
When I modified the APIs I brought Javascript to match the API defined in Java.  I utilize the definition in my test code, but agree, I never actually utilize it within the EccCryptoContext.

Thinking forward, if we add other use cases for ECC (encrypt/decrypt) then it would have value.  That said, the argument "this might have value in the future" is probably not a good enough reason to add the extra complexity - we should add it in when we actually need it.

I see no problem with removing the mode parameter.